From 4baac8ce815225a3fe8fbfdaa06d114d5e7aa144 Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Thu, 9 Dec 2021 12:41:17 +0100
Subject: [PATCH] Fixed build against OpenSSL 3.0.

Deprecated function FIPS_mode_set() was removed in OpenSSL 3.0.

Switched to EVP_default_properties_enable_fips() as described in OpenSSL
3.0 migration guide.

Signed-off-by: Vitaly Zaitsev <vitaly@easycoding.org>
---
 Telegram/SourceFiles/core/utils.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Telegram/SourceFiles/core/utils.cpp b/Telegram/SourceFiles/core/utils.cpp
index 160af3f77047..16b9e2d8fa51 100644
--- a/Telegram/SourceFiles/core/utils.cpp
+++ b/Telegram/SourceFiles/core/utils.cpp
@@ -125,7 +125,11 @@ namespace ThirdParty {
 	void finish() {
 		CRYPTO_cleanup_all_ex_data();
 #ifndef LIBRESSL_VERSION_NUMBER
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+		EVP_default_properties_enable_fips(nullptr, 0);
+#else
 		FIPS_mode_set(0);
+#endif
 #endif
 		ENGINE_cleanup();
 		CONF_modules_unload(1);
